sdks/java/core/src/main/java/org/apache/beam/sdk/coders/StandardCoder.java

```diff
@@ -228,4 +232,10 @@ public Object structuralValue(T value) throws Exception {
       }
     }
   }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public TypeDescriptor<T> getEncodedTypeDescriptor() {
+    return (TypeDescriptor<T>) DEFAULT_TYPE_DESCRIPTOR;
```

Comment:
It is a good point, for example `AvroCoder<T>` and `SerializableCoder<T>` have this relationship with their superclass (though they are `AtomicCoder`). I think most such coders are, themselves, parameterized on the class, so they should be able to get a good / OK type descriptor. Is there another sort of example?

If you don't mind switching back, I actually would prefer it - and maybe the type variable will even give a user a hint where to look.